db/event_manager_test.go

```diff
@@ -748,5 +739,28 @@ func TestUnavailableWebhook(t *testing.T) {
 		em.RaiseDocumentChangeEvent(body, "", channels)
 	}
 
-	time.Sleep(50 * time.Millisecond)
+}
+
+// assertChannelLengthWithTimeout will return the number of items seen in the channel, along with an error if it timed out.
+// WARNING: This function will drain the channel of items!
+func assertChannelLengthWithTimeout(c chan Body, expectedLength int, timeout time.Duration) (length int, err error) {
```

Comment:
Maybe poorly named, as it doesn't really assert the channel length... Couldn't think of a better one besides `countChannelItemsWithTimeout`, but if you have any suggestions I'd be thankful.

Using the `expectedLength` as a shortcut perhaps confuses things, but I didn't want to have the timeout be a time penalty for the function actually returning on the happy path.